3.4 Release Notes (#6342)
influxdata/docs-v2: content/shared/v3-core-enterprise-release-notes/_index.md

> All updates to Core are automatically included in Enterprise.
> The Enterprise sections below only list updates exclusive to Enterprise.

## v3.4.0 {date="2025-08-27"}

### Core

#### Features

- **Token Provisioning**:
  - You can now generate admin tokens offline that can then be loaded up by the database at runtime if the tokens do not already exist, which works great for automated deployments and containerized environments. ([#26734](https://github.com/influxdata/influxdb/pull/26734))
- **Azure Endpoint**:
  - You can now specify the Azure endpoint for object-store connections using the `--azure-endpoint` parameter ([#26687](https://github.com/influxdata/influxdb/pull/26687))
- **No_Sync via CLI**:

>   - The no_sync option can now be specified with the `--no-sync` parameter via the CLI on write commands ([#26703](https://github.com/influxdata/influxdb/pull/26703))

Reviewer: ```suggestion
  - Use the `--no-sync` option with `influxdb3 write` to skip waiting for WAL persistence on write and immediately return a response to the write request. ([#26703](https://github.com/influxdata/influxdb/pull/26703))
```

  
#### Bug Fixes

> - Tag and field names are validated when creating tables ([#26641](https://github.com/influxdata/influxdb/pull/26641))

Reviewer: ```suggestion
- Validate tag and field names when creating tables ([#26641](https://github.com/influxdata/influxdb/pull/26641))
```

- Using GROUP BY twice on the same column no longer causes incorrect data ([#26732](https://github.com/influxdata/influxdb/pull/26732))

#### Security & Misc

> - Overly verbose TableIndexCache log is now quiet ([#26709](https://github.com/influxdata/influxdb/pull/26709))
> - WAL replay concurrency limit now defaults to number of CPU cores, preventing possible OOM ([#26715](https://github.com/influxdata/influxdb/issues/26716))
> - Unsafe signal_handler code removed ([#26685](https://github.com/influxdata/influxdb/pull/26685))
> - Python version upgraded to 3.13.7-20250818 ([#26686](https://github.com/influxdata/influxdb/pull/26686), [#26700](https://github.com/influxdata/influxdb/pull/26700))
> - Fixed an issue where tags with / in the name broke the primary key

Reviewer: ```suggestion
- Reduce verbosity of the TableIndexCache log. ([#26709](https://github.com/influxdata/influxdb/pull/26709))
- WAL replay concurrency limit defaults to number of CPU cores, preventing possible OOMs. ([#26715](https://github.com/influxdata/influxdb/issues/26716))
- Remove unsafe signal_handler code. ([#26685](https://github.com/influxdata/influxdb/pull/26685))
- Upgrade Python version to 3.13.7-20250818. ([#26686](https://github.com/influxdata/influxdb/pull/26686), [#26700](https://github.com/influxdata/influxdb/pull/26700))
- Tags with `/` in the name no longer break the primary key.
```

### Enterprise

All Core updates are included in Enterprise. Additional Enterprise-specific features and fixes:

#### Features

- **Token Provisioning**:

>   - In addition to admin token, you can now generate resource tokens offline that can then be loaded up by the database at runtime.

Reviewer: ```suggestion
  - Generate _resource_ and _admin_ tokens offline and use them when starting the database.
```

> - It's now possible to select a home or trial license without using an interactive terminal. Add a `--license-type` [home | trial | commercial] argument to the `serve` command to automate the selection of the license type.

Reviewer: ```suggestion
- Select a home or trial license without using an interactive terminal.
  Use `--license-type` [home | trial | commercial] option to the `influxdb3 serve` command to automate the selection of the license type.
```

#### Bug Fixes

> - Fixed an issue causing the Processing Engine to be initialized regardless of the specified `--mode`.

Reviewer: ```suggestion
- Don't initialize the Processing Engine when the specified `--mode` does not require it.
```

> - Fixed a panic when `INFLUXDB3_PLUGIN_DIR` was set in containers without the Processing Engine enabled

Reviewer: ```suggestion
- Don't panic when `INFLUXDB3_PLUGIN_DIR` is set in containers without the Processing Engine enabled.
```
